torch/_dynamo/eval_frame.py

```diff
@@ -113,7 +116,27 @@ def _maybe_set_eval_frame(callback: DynamoCallback):
         )
         return callback
     else:
-        return set_eval_frame(callback)
+        assert state in ["enter", "exit"]
+        if torch._dynamo.compiled_autograd.in_compiled_autograd_region:
+            # NOTE: Compiled Autograd graph cannot be executed in eager mode and must be Dynamo traced through
+            return set_eval_frame(callback)
+        else:
+            if context_id not in context_id_to_warmup_count:
+                context_id_to_warmup_count[context_id] = 0
```

Comment:
If you use a `defaultdict` then you don't need to separately check for and insert `0` here.